src/error-snippets/index.ts

```diff
@@ -0,0 +1,54 @@
+import { codeFrameColumns } from "@babel/code-frame";
+import { findRelevantStackFrame, resolveWithStackFrame } from "./frames";
+import { extractSourceMaps, resolveWithSourceMap } from "./source-maps";
+import { getSourceCodeFile, formatFileNameHeader } from "./utils";
+import logger from "../utils/logger";
+import { SNIPPET_LINES_ABOVE, SNIPPET_LINES_BELOW } from "./constants";
+import type { ResolvedFrame, SufficientStackFrame, WithSnippetError } from "./types";
+
+const stackFrameResolver = async (stackFrame: SufficientStackFrame): Promise<ResolvedFrame> => {
+    const fileContents = await getSourceCodeFile(stackFrame.fileName);
+    const sourceMaps = await extractSourceMaps(fileContents, stackFrame.fileName);
+
+    return sourceMaps ? resolveWithSourceMap(stackFrame, sourceMaps) : resolveWithStackFrame(stackFrame, fileContents);
+};
+
+export const extendWithCodeSnippet = async (err: WithSnippetError): Promise<WithSnippetError> => {
```

Comment:
this function does not throw exceptions either.
The worst it can do - log a warning into console.